### PR TITLE
Add deploy key functionality

### PR DESCRIPTION
**Pull request by ZPascal**
The PR includes the basic functionality and a fix of the failed master validation run: https://github.com/ad-m/github-push-action/actions/runs/2269331602

**Tested by the following jobs:**
- [no-ssh](https://github.com/ZPascal/grafana_dashboard_templater/runs/6290902167?check_suite_focus=true)
- [ssh](https://github.com/ZPascal/grafana_dashboard_templater/runs/6291050229?check_suite_focus=true)

### CODE REVIEW

**Comment by ZPascal** [comment]
Hi @ad-m or @jackton1, the PR is ready for a review :) Could you please review it?

**Comment by ZPascal** [comment]
Is @jackton1 the PR from your side ready to merge?

**Review by jackton1 — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -39,6 +39,31 @@ jobs:
...
+An example workflow to authenticate with GitHub Platform via Deploy Keys or in general SSH:
+
+```yaml
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        ssh-key: ${{ secrets.SSH_PRIVATE_KEY }}
```
Comment (jackton1) [inline README.md:51]: I'll also suggest highlighting that the `persist-credentials` input should be set to `true`

File: start.sh [hunk 2]
```diff
@@ -14,7 +15,7 @@ echo "Push to branch $INPUT_BRANCH";
 };
 
 if ${INPUT_FORCE}; then
-    _FORCE_OPTION='--force'
+    _FORCE_OPTION='--force-with-lease'
```
[the remark this answers is not in the document]
Comment (ZPascal) [reply start.sh:18]: Yes, I'll handle that.

File: README.md (hunk 1 above)
Comment (ZPascal) [reply README.md:51]: Sure, that should not result in a problem.

File: start.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (ZPascal) [reply start.sh]: Done

File: README.md (hunk 1 above)
Comment (ZPascal) [reply README.md:51]: Done

**Review by jackton1 — APPROVED** [review]

**Pull request merged**